src/axom/sidre/core/MFEMSidreDataCollection.hpp

```diff
@@ -457,6 +456,16 @@ class MFEMSidreDataCollection : public mfem::DataCollection
   // This is stored for convenience.
   Group* m_named_bufs_grp;
 
+  // Used to retain ownership of components of reconstructed Meshes and GridFuncs
+  // Instead of using flags to keep track of ownership between this class
+  // and the mfem::DataCollection subobject, always have the subobject
+  // retain a non-owning pointer and manage memory via unique_ptr in
+  // this class for consistency
+  std::unique_ptr<mfem::Mesh> m_owned_mesh;
```

Comment:
I'd like to hear more about some of these use cases - though from my understanding the existing interface will be preserved in that `mfem::DataCollection::mesh` will remain a non-owning `Mesh*` that references the `unique_ptr` when the mesh is owned by the datastore, and an external mesh otherwise.